### PR TITLE
Fix #349: JS generation should check the validity of all levels in a …

### LINKED ISSUE #349: JS generation: need to check validity of multi-level fields
JS generation uses if(typeof field != undefined) before accessing a field, but if field is multilevel, then it fails if the first level isn't defined, for instance if(typeof x.y != undefined) fails if x isn't defined.

### PR DESCRIPTION
**Pull request by bserdar**
…field name

### CODE REVIEW

File: mongo/src/main/java/com/redhat/lightblue/mongo/crud/js/IfStatement.java [hunk 1]
```diff
@@ -28,11 +28,27 @@ public IfStatement(Expression test,Statement...s) {
     }
 
     public static IfStatement ifDefined(Name var,Statement...s) {
-        return new IfStatement(new SimpleExpression("typeof this.%s != 'undefined'", var),s);
+        StringBuffer b=new StringBuffer(64);
+        for(int i=0;i<var.length();i++) {
+        }
```
Comment (paterczm) [inline mongo/src/main/java/com/redhat/lightblue/mongo/crud/js/IfStatement.java]: What is this empty loop for?

**Comment by bserdar** [comment]
Leftover from refactoring. Removed.

File: mongo/src/test/java/com/redhat/lightblue/mongo/crud/TranslatorTest.java [hunk 2]
```diff
@@ -326,11 +326,11 @@ public void translateNullReference() throws Exception {
     @Test
     public void translateJS() throws Exception {
         DBObject obj = expressionTranslator.translate(md, query("{'field':'field7.*.elemf1','op':'=','rfield':'field7.*.elemf2'}"));
-        Assert.assertEquals("function() {var r0=false;{if(typeof this.field7 != 'undefined'){for(var ri1=0;ri1<this.field7.length;ri1++){if(typeof this.field7 != 'undefined'){for(var ri2=0;ri2<this.field7.length;ri2++){r0=field7[ri1].elemf1 == field7[ri2].elemf2;if(r0){break;}}}if(r0){break;}}}}return r0;}",
+        Assert.assertEquals("function() {var r0=false;{if(typeof this.field7!='undefined'){for(var ri1=0;ri1<this.field7.length;ri1++){if(typeof this.field7!='undefined'){for(var ri2=0;ri2<this.field7.length;ri2++){r0=field7[ri1].elemf1 == field7[ri2].elemf2;if(r0){break;}}}if(r0){break;}}}}return r0;}",
```
Comment (dcrissman) [inline mongo/src/test/java/com/redhat/lightblue/mongo/crud/TranslatorTest.java]: Is it worth touching this class just for spacing changes?

**Comment by bserdar** [comment]
The test compares with spaces, so I had to fix the string to match the
generated string.

**Pull request merged**